packages/components/pie-button/test/component/pie-button.spec.ts

```diff
@@ -272,422 +156,228 @@ test.describe('Form Actions', () => {
-            test.describe('when "responsiveSize" is "expressive"', () => {
-                test('the button should have the expected attribute', async ({ mount }) => {
-                    const props: ButtonProps = {
-                        size: 'xsmall',
-                        isResponsive: true,
-                        responsiveSize: 'expressive',
-                    };
-
-                    const component = await mount(
-                        PieButton,
-                        {
-                            props,
-                            slots: {
-                                default: 'Click me!',
-                            },
-                        },
-                    );
-
-                    await expect(component.locator('button'))
-                        .toHaveClass([/o-btn--expressive/]);
-                });
+                // Assert
+                await expect(buttonComponent.componentLocator.locator('button')).not.toHaveClass([/o-btn--productive/, /o-btn--responsive/]);
             });
         });
 
-        test.describe('when "responsiveSize" is "productive"', () => {
-            test('the button should have the expected attribute', async ({ mount }) => {
-                const props: ButtonProps = {
-                    size: 'xsmall',
-                    isResponsive: true,
-                    responsiveSize: 'productive',
-                };
-
-                const component = await mount(
-                    PieButton,
-                    {
-                        props,
-                        slots: {
-                            default: 'Click me!',
-                        },
-                    },
-                );
-
-                await expect(component.locator('button'))
-                    .toHaveClass(/o-btn--productive/);
+        const responsiveSizeTestCases = [
+            {
+                size: 'xsmall', responsiveSize: 'expressive', expectedClass: [/o-btn--expressive/], isResponsive: true,
+            },
+            {
+                size: 'xsmall', responsiveSize: 'productive', expectedClass: [/o-btn--productive/], isResponsive: true,
+            },
+        ];
+
+        responsiveSizeTestCases.forEach((testCase) => {
+            test.describe('when "isResponsive" is true', () => {
+                test.describe(`when "responsiveSize" is "${testCase.responsiveSize}"`, () => {
+                    test('the button should have the expected attribute', async ({ page }) => {
+                        // Arrange
+                        const buttonComponent = new ButtonComponent(page, 'button--primary');
+                        await buttonComponent.load({ ...testCase });
+
+                        // Assert
+                        await expect(buttonComponent.componentLocator.locator('button'))
+                        .toHaveClass(testCase.expectedClass);
+                    });
+                });
```

Comment:
When you get around to your test audit, I think we should question testing for the presence of particular classes

Reply:
Removed this test as discussed as we have visual test coverage.